File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxSetTypingExceptionTest.java
package im.tox.tox4j.core.exceptions;

import im.tox.tox4j.ToxCoreTestBase;
import im.tox.tox4j.core.ToxCore;
import org.junit.Test;
import scala.runtime.BoxedUnit;

import static org.junit.Assert.assertEquals;

public class ToxSetTypingExceptionTest extends ToxCoreTestBase {

  @Test
  public void testSetTypingToNonExistent() throws Exception {
    try (ToxCore<BoxedUnit> tox = newTox()) {
      addFriends(tox, 1);
      try {
        tox.setTyping(1, true);
        fail();
      } catch (ToxSetTypingException e) {
        assertEquals(ToxSetTypingException.Code.FRIEND_NOT_FOUND, e.code());
      }
    }
  }

}

File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxSetTypingExceptionTest.scala
package im.tox.tox4j.core.exceptions

import im.tox.tox4j.testing.ToxTestMixin
import org.scalatest.FunSuite

final class ToxSetTypingExceptionTest extends FunSuite with ToxTestMixin {

  test("SetTypingToNonExistent") {
    interceptWithTox(ToxSetTypingException.Code.FRIEND_NOT_FOUND)(
      _.setTyping(1, typing = true)
    )
  }

}
